src/YahooFinance.py

```diff
@@ -198,9 +198,44 @@ def parse_modules(self, content):
     results = data.get('quoteSummary', {}).get('result', None)
     if not results:
       logging.error('Could not parse response for url: ' + self.url)
-      return
+      return False
     for module in self.modules:
       for result in results:
         if module in result:
           self.module_data[module] = result[module]
-          break
\ No newline at end of file
+          break
+    return True
+
+  def _get_balance_sheet_history(self, position):
+    history = []
+    for stmt in self.module_data.get('balanceSheetHistory').get('balanceSheetStatements'):
+      history.append(stmt.get(position).get('raw'))
+    return history
+
+  def _get_income_statement_history(self, position):
+    history = []
+    for stmt in self.module_data.get('incomeStatementHistory').get('incomeStatementHistory'):
+      history.append(stmt.get(position).get('raw'))
+    return history
+
+  def _get_roic_history(self):
+    net_income_history = self._get_income_statement_history('netIncome')
+    cash_history = self._get_balance_sheet_history('cash')
+    long_term_debt_history = self._get_balance_sheet_history('longTermDebt')
+    stockholder_equity_history = self._get_balance_sheet_history('totalStockholderEquity')
+    roic_history = []
+    for i in range(0, len(net_income_history)):
+      roic_history.append(
+        (
+          net_income_history[i]
+          /
+          (stockholder_equity_history[i] + long_term_debt_history[i] - cash_history[i])
```

Comment:
suggestion: Can we pull this ROIC calculation in to a helper function? It's not specific to the Yahoo module and could then be unit tested individually as well.

Perhaps we can just put it as part of the `RuleOneInvestingCalculations.py`.

What do you think about that idea?

Reply:
Thanks for all your suggestions! I'll come back to this sometime this week.

Reply:
Done.